src/jwplayer/players/DesktopArticleVideoPlayer/DesktopArticleVideoPlayer.tsx

```diff
@@ -90,32 +90,19 @@ export const DesktopArticleVideoPlayerContent: React.FC<DesktopArticleVideoPlaye
 					>
 						{jwpAdsSetupComplete.strategyRulesEnabled ? (
 							<div>
-								<div className={styles.topBar}>
-									{!isScrollPlayer && <UnmuteButton />}
-									{isScrollPlayer && (
-										<CloseButton dismiss={() => setDismissed(true)} iconColor={'#fff'} className={styles.closeButton} />
-									)}
-								</div>
+								<TopBar isScrollPlayer={isScrollPlayer} onClickClose={() => setDismissed(true)} ref={topBarRef} />
 								<StrategyRulesWrapper
 									getDismissed={getDismissed}
 									config={getArticleVideoConfig(videoDetails)}
 									onReady={onPlayerInstanceReady}
+									topBarRef={topBarRef}
```

Comment:
Instead of passing ref and setting a class name you could conditionally render TopBar component based on `isPlayerReady` variable and `player` from PlayerContext.

```
const [isPlayerReady, setIsPlayerReady] = useState(false);
const { player } = useContext(PlayerContext);
const shouldRenderTopBar = isPlayerReady && !!player;
```

```
{shouldRenderTopBar && <TopBar... />}
```

Reply:
I was thinking about this, but I was not sure whether it's reactive or not. Thank you, I'll give it a try.

Reply:
It worked! I need to understand the context API better. It seems to be similar to Vue's provide/inject concept.

Reply:
everything that is based on props or state is "reactive" in React ;)